### PR TITLE
Ll insertions

### PR DESCRIPTION
**Pull request by wosunkwo**

### CODE REVIEW

**Comment by wosunkwo** [comment]
- [x] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
     - [x] “Happy Path” - Expected outcome
     - [x] Expected failure
     - [x] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Link to code
     - [x] Picture of whiteboard

File: code401-challenges/src/main/java/code401/challenges/linkedlist/LinkedList.java [hunk 1]
```diff
@@ -50,17 +50,98 @@ public boolean includes(int value) {
...
-        while (current != null) {
-            resultArr[arrCnt] = current.value;
-            current = current.next;
-            arrCnt++;
+    }
+
+
+    //This function adds a new value to the end of the linked list
+    public void append(int value){
+        Node current = new Node();
```
Comment (evanslaton) [inline code401-challenges/src/main/java/code401/challenges/linkedlist/LinkedList.java:76]: You could also make your Node constructor take in an argument that it sets as the value

File: code401-challenges/src/main/java/code401/challenges/linkedlist/LinkedList.java [hunk 2]
```diff
@@ -50,17 +50,98 @@ public boolean includes(int value) {
...
+            while(iterator.next != null){
+                iterator = iterator.next;
+            }
+            iterator.next = current;
+        }
+        this.size++;
+    }
+
+    //this function adds a new value at the back of the specified value
+    public int insertBefore(int value, int newVal){
```
Comment (evanslaton) [inline code401-challenges/src/main/java/code401/challenges/linkedlist/LinkedList.java:92]: I'm not sure having this method return an int is 100% necessary

File: code401-challenges/src/main/java/code401/challenges/linkedlist/LinkedList.java [hunk 3]
```diff
@@ -50,17 +50,98 @@ public boolean includes(int value) {
...
+                    current = current.next;
+                }
+            }
+        }
+        System.out.println("The linked list does not contain "+ value);
+        return -1;
+    }
+
+    //this function adds a new value in front of a specified value
+    public int insertAfter(int value, int newVal){
```
Comment (evanslaton) [inline code401-challenges/src/main/java/code401/challenges/linkedlist/LinkedList.java:125]: Same as previous comment

File: code401-challenges/src/test/java/code401/challenges/linkedlist/LinkedListTest.java [hunk 4]
```diff
@@ -67,4 +66,88 @@ public void testForReturnFalse(){
...
         assertFalse("This should return false if the linked list does not contains the value we are looking for ", classUnderTest.includes(4));
     }
+
+    //************************************ Test for Code Challenge 6 ********************
+
+    // Test to see if my append method successfully adds a new value to the end of my linked list
+    @Test
+    public void testForAddAtEndOfList(){
+        LinkedList classUnderTest = new LinkedList();
+        classUnderTest.insert(4);
```
Comment (evanslaton) [inline code401-challenges/src/test/java/code401/challenges/linkedlist/LinkedListTest.java:76]: Why not just use only `append()` to create this LinkedList?
